classes/robot/crawler.php

```diff
@@ -143,6 +143,8 @@ public function absolute_url($base, $rel) {
         $host = $parts['host'];
 
         // Remove non-directory element from path.
+        // This is converting anything like '/XYZ' to '' except of '//' and '///' and '//////' and etc.
+        // It seems to be wrong.
```

Comment:
why does this look wrong to you? It looks fine to me

If you think it is wrong then write a failing unit test